### PR TITLE
Use COCO2017

### PR DESCRIPTION
**Pull request by yuyu2172**
https://github.com/chainer/chainercv/issues/651

### CODE REVIEW

**Comment by Hakuyume** [comment]
How to use 'train', 'val', 'minival', 'valminusminival' of 2014? If there are any correspondences, it would be nice to write them.

**Comment by yuyu2172** [comment]
Is saying correspondence in doc enough?
`train + valminusminival of 2014 == train of 2017` and `minival of 2014 == val of 2017`

File: chainercv/datasets/coco/coco_instance_segmentation_dataset.py [hunk 1]
```diff
@@ -56,6 +48,10 @@ class COCOInstanceSegmentationDataset(GetterDataset):
             under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/coco`.
         split ({'train', 'val', 'minival', 'valminusminival'}): Select
             a split of the dataset.
+        year ({'2014', '2017'}): Use a dataset prepared for a challenge
+            held in :obj:`year`.
```
Comment (Hakuyume) [inline chainercv/datasets/coco/coco_instance_segmentation_dataset.py]: The competition that used COCO 2014 was held in 2015. http://cocodataset.org/#detection-2015

Comment (yuyu2172) [reply chainercv/datasets/coco/coco_instance_segmentation_dataset.py]: thanks

**Comment by Hakuyume** [comment]
I marked this PR as `no-compat` since the default value of `year` is `2017`.

**Comment by yuyu2172** [comment]
COCO was never included in any release, so this is not no-compat

**Comment by Hakuyume** [comment]
I see. I removed the label.

File: tests/datasets_tests/coco_tests/test_coco_instance_segmentation_dataset.py [hunk 2]
```diff
@@ -9,18 +9,36 @@
...
+        'use_crowded': [False, True],
+        'return_crowded': [False, True],
+        'return_area': [False, True]})
+    params = testing.product_dict(
+        split_years,
+        use_and_return_args)
+    return params
+
+
+@unittest.skipUnless(_available, 'pycocotools is not installed')
```
Comment (Hakuyume) [inline tests/datasets_tests/coco_tests/test_coco_instance_segmentation_dataset.py]: Do we need this condition?

Comment (Hakuyume) [reply tests/datasets_tests/coco_tests/test_coco_instance_segmentation_dataset.py]: Sorry, this is my mistake.

**Pull request merged**